engine/api/integration/dao_model.go

```diff
@@ -147,22 +131,46 @@ func ModelExists(db gorp.SqlExecutor, name string) (bool, error) {
 
 // InsertModel inserts a integration model in database
 func InsertModel(db gorp.SqlExecutor, m *sdk.IntegrationModel) error {
-	dbm := integrationModel(*m)
-	if err := db.Insert(&dbm); err != nil {
+	givenPublicConfig := m.PublicConfigurations.Clone()
+	dbm := integrationModel{IntegrationModel: *m}
+	if err := gorpmapping.InsertAndSign(context.Background(), db, &dbm); err != nil {
 		return sdk.WrapError(err, "Unable to insert integration model %s", m.Name)
 	}
-	*m = sdk.IntegrationModel(dbm)
+	*m = dbm.IntegrationModel
+	m.PublicConfigurations = givenPublicConfig
+	m.PublicConfigurations.Blur()
 	return nil
 }
 
 // UpdateModel updates a integration model in database
 func UpdateModel(db gorp.SqlExecutor, m *sdk.IntegrationModel) error {
-	dbm := integrationModel(*m)
-	if n, err := db.Update(&dbm); err != nil {
+	// reload the previous config to encuse we don't store placeholder
+	var oldModel sdk.IntegrationModel
+
+	givenPublicConfig := m.PublicConfigurations.Clone()
+	for k := range givenPublicConfig {
+		for kk, cfg := range givenPublicConfig[k] {
+			if cfg.Type == sdk.IntegrationConfigTypePassword && cfg.Value == sdk.PasswordPlaceholder {
+				if oldModel.ID == 0 {
+					var err error
+					oldModel, err = LoadModel(db, m.ID, true)
+					if err != nil {
+						return err
+					}
+				}
+				cfg.Value = oldModel.PublicConfigurations[k][kk].Value
```

Comment:
Maybe this should be strange and impossible but can the oldModel public config not contains all the keys that are in the updated model ? In other words should be check that oldModel.PublicConfigurations[k] is not nil ?